### ISSUE TITLE
background color not respected in floating windows

### ISSUE THREAD

**Issue by wookayin** [voldikss/vim-floaterm#11]
I can see the highlight group was applied like below, but actually it does not change the background of the floating windows.

```
FloatTermNormal xxx guibg=red
```

Because `setwinvar` is being called on the terminal buffer, not a floating window one:

```
call setwinvar(bufnr('%'), '&winhl', 'Normal:FloatTermNormal')
```

Actually `nvim_open_win` returns a different window number and this is the correct window handle to call setwinvar on.

**Comment by voldikss** [comment]
Should I use `setbufvar`?

Jongwook Choi <notifications@github.com> 于 2019年10月19日周六 下午3:48写道：

**Comment by voldikss** [comment]
Fixed, thanks!

**Comment by voldikss** [comment]
Fixed, thanks!

**Comment by wookayin** [comment]
Thanks, it works. But do you have any idea on how can we deal with the leftmost column whose background color is different?

![image](https://user-images.githubusercontent.com/1009873/67152636-9f25e880-f2a8-11e9-9ad3-2d4bf91889eb.png)

**Comment by voldikss** [comment]
> But do you have any idea on how can we deal with the leftmost column whose background color is different?

The color is controlled by `hi-FoldColumn`. I had `set foldcolumn=2` in order to add padding to the floating window. But it seems annoying...

**Comment by voldikss** [comment]
The first solution is setting the `FoldColumn`'s `guibg` option the same as `NormalFloat`'s. However, `FoldColumn` in other windows/buffers will also be changed.

Another method is `set foldcolumn=0`, but there are no other ways to add a padding to the floating window...

**Comment by wookayin** [comment]
Thanks. The following works for me (no effects on other windows/buffers). However the margin is only applied in the leftmost column, not the rightmost column -- a bit ugly. but that would be fine.


```
-    call setbufvar(bufnr('%'), '&winhl', 'Normal:FloatTermNormal')
+call 
setbufvar(bufnr('%'), '&winhl', 'Normal:FloatTermNormal,FoldColumn:FloatTermNormal')
```

FYI [This reddit thread](https://www.reddit.com/r/neovim/comments/di41cm/a_floating_terminal_with_borderpadding/) discusses how to have a margin. It is quite hacky since two floating windows are created -- I could contribute a PR that introduces a similar feature.

**Comment by voldikss** [comment]
> The following works for me (no effects on other windows/buffers)

I mean if you set `foldcolumn` option to a positive number in another window, the `FoldColumn`' color in that window would also be `FloatTermNormal`'s color.

> FYI This reddit thread discusses how to have a margin. It is quite hacky since two floating windows are created -- I could contribute a PR that introduces a similar feature.

Yes I've read that post. Very Jacky indeed. Thank you for you help with so many bugs fixes. Look forward to your PR.

**Comment by wookayin** [comment]
Closing this as fixed by 0225e54.
